plugins/modules/aci_l3out_to_sr_mpls_infra_l3out.py

```diff
@@ -0,0 +1,355 @@
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    "metadata_version": "1.1",
+    "status": ["preview"],
+    "supported_by": "certified",
+}
+
+DOCUMENTATION = r"""
+---
+module: aci_l3out_to_sr_mpls_infra_l3out
+short_description: Manage Layer 3 Outside (L3Out) to SR-MPLS Infra L3Outs objects (l3ext:ConsLbl)
+description:
+- Manage Layer 3 Outside (L3Out) to SR-MPLS Infra L3Outs objects on Cisco ACI fabrics.
+options:
+  tenant:
+    description:
+    - The name of an existing tenant.
+    type: str
+    aliases: [ tenant_name ]
+  l3out:
+    description:
+    - The name of an existing SR MPLS VRF L3Out.
+    type: str
+    aliases: [ l3out_name, name ]
+  infra_l3out:
+    description:
+    - The name of an existing SR-MPLS Infra L3Out.
+    type: str
+    aliases: [ infra_l3out_name ]
+  external_epg:
+    description:
+    - The distinguished name (DN) of the external EPG.
+    type: str
+    aliases: [ external_epg_dn ]
+  outbound_route_map:
+    description:
+    - The distinguished name (DN) of the outbound route map.
+    type: str
+    aliases: [ outbound_route_map_dn, outbound ]
+  inbound_route_map:
+    description:
+    - The distinguished name (DN) of the inbound route map.
+    - Use an empty string to remove the inbound route map.
+    type: str
+    aliases: [ inbound_route_map_dn, inbound ]
+  description:
+    description:
+    - Description for the L3Out.
+    type: str
+    aliases: [ descr ]
+  state:
+    description:
+    - Use C(present) or C(absent) for adding or removing.
+    - Use C(query) for listing an object or multiple objects.
+    type: str
+    choices: [ absent, present, query ]
+    default: present
+  name_alias:
+    description:
+    - The alias for the current object. This relates to the nameAlias field in ACI.
+    type: str
+extends_documentation_fragment:
+- cisco.aci.aci
+- cisco.aci.annotation
+- cisco.aci.owner
+
+notes:
+- The C(tenant) and C(l3out) used must exist before using this module in your playbook.
+- The M(cisco.aci.aci_tenant) and M(cisco.aci.aci_l3out) modules can be used for this.
+seealso:
+- module: cisco.aci.aci_tenant
+- module: cisco.aci.aci_l3out
+- name: APIC Management Information Model reference
+  description: More information about the internal APIC class B(l3ext:ConsLbl).
+  link: https://developer.cisco.com/docs/apic-mim-ref/
+author:
+- Akini Ross (@akinross)
+"""
+
+EXAMPLES = r"""
+- name: Add a new l3out to sr-mpls infra l3out
+  cisco.aci.aci_l3out_to_sr_mpls_infra_l3out:
+    host: apic
+    username: admin
+    password: SomeSecretPassword
+    tenant: production
+    l3out: prod_l3out
+    description: L3Out for Production tenant
+    infra_l3out: infra_l3out_name
+    external_epg: uni/tn-production/out-l3out_name/instP-external_epg_name
+    outbound_route_map: uni/tn-production/prof-outbound_route_map_name
+    inbound_route_map: uni/tn-production/prof-inbound_route_map_name
```

Comment:
is this the only way for the users to give inputs?
or will this be changed when the other supporting PR is merged?

Reply:
disscussed and resolved